cg/meta/workflow/balsamic.py

```diff
@@ -604,30 +604,35 @@ def get_pipeline_version(self, case_id: str) -> str:
         )
         return config_data["analysis"]["BALSAMIC_version"]
 
-    def family_has_correct_number_tumor_normal_samples(self, case_id: str) -> bool:
+    def case_has_correct_number_tumour_normal_samples(self, case_internal_id: str) -> bool:
         """Evaluates if a case has exactly one tumor and up to one normal sample in ClinicalDB.
         This check is only applied to filter jobs which start automatically"""
-        query = (
-            self.status_db.query(Sample)
-            .join(Family.links, FamilySample.sample)
-            .filter(Family.internal_id == case_id)
-            .filter(Family.data_analysis == self.pipeline)
-        )
-        return all(
-            [
-                len(query.filter(Sample.is_tumour == False).all()) <= 1,
-                len(query.filter(Sample.is_tumour == True).all()) == 1,
-            ]
+
+        samples: List[Sample] = self.status_db.get_samples_by_case_and_pipeline(
+            case_internal_id=case_internal_id, pipeline=self.pipeline
         )
 
+        normal_samples_count: int = self.count_normal_samples(samples=samples)
```

Comment:
I think we should skip type hints when the type is super clear by context, for instance here.

Reply:
I think it is still good to do to tell non-humans (like editors and mypy (we should add it)) what to expect